### PR TITLE
Add CLion instructions for windows

### PR DESCRIPTION
**Pull request by robertapplin**
**Description of work.**
This PR adds a new developer documentation page for setting up CLion on Linux and Windows. It moves previously existing Linux documentation to this page, and expands on the instructions to help set it up on a Windows environment.

**To test:**
Build the dev docs target and make sure the links on the documentation work.

Read through the documentation and try set up CLion on windows to make sure it makes sense

Part of #34490 

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: dev-docs/source/CLion.rst [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+- Create a new ``System`` toolchain using the ``+`` icon and call it ``Default``
+- Edit the CMake field to point to your conda installed ``cmake``, e.g. ``/path/to/mambaforge/envs/mantid-developer/bin/cmake``
+- Edit the Build Tool field to point to your conda installed ``ninja``, e.g. ``/path/to/mambaforge/envs/mantid-developer/bin/ninja``
+- On Linux, ``Let CMake detect`` the C Compiler and C++ Compiler.
+- On Windows, the C Compiler and C++ Compiler should be directed at the same ``cl.exe`` in your Visual Studio installation, e.g. ``C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.29.30133/bin/Hostx64/x64/cl.exe``
+
+To set up CMake:
+
+- Navigate to ``File > Settings > Build, Execution, Deployment > CMake``
+- On Linux, set your Build type to be ``Debug``. On Windows, set your build type to be ``DebugWithRelRuntime``
```
Comment (sf1919) [inline dev-docs/source/CLion.rst:36]: Could we add a note that if the option isn't available you just type it in?

**Comment by robertapplin** [comment]
Set the Debugger to be Bundled LLDB for windows

File: dev-docs/source/CLion.rst [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+- Start CLion using the above steps
+- Navigate to ``File > Settings > Tools > Terminal``
+- To the end of the ``Shell path`` option, add ``--rcfile ~/.clionrc``
+
+Debugging with CLion
+####################
+
+To debug workbench, you'll need to edit the ``workbench`` CMake Application configuration.
+
+- Set the executable to be the ``python.exe`` in your conda installation, e.g. ``/path/to/mambaforge/envs/mantid-developer/bin/python.exe``
```
Comment (sf1919) [inline dev-docs/source/CLion.rst:72]: In windows the `python.exe` file is not in the bin folder but the level above that.

**Comment by robertapplin** [comment]
Thanks @sf1919 . I have done some re-formatting to hopefully make it easier to read as well, so it might be worth build the `dev-docs-html` target again and having a look

File: dev-docs/source/CLion.rst [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+Please note that these instructions only work when using a Ninja generator from a Windows or Linux operating system.
+
+You will also need to have Visual Studio installed on windows.
+
+If you haven't installed CLion yet do that now, CLion can be installed from `here <https://jetbrains.com/clion/download/>`_.
+
+Opening CLion
+#############
+
+The first time you build from CLion, you will most likely need to launch it from a terminal or command line to make sure you have access to all the relevent tools.
```
Comment (sf1919) [inline dev-docs/source/CLion.rst:22]:
```suggestion
The first time you build from CLion, you will most likely need to launch it from a terminal or command line to make sure you have access to all the relevant tools.
```

File: dev-docs/source/CLion.rst [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+      - On Windows: ``--preset=win-ninja``
+
+#. Set the build directory to the ``build`` directory if it is not the default (you'll need to use the full path if its outside the source directory)
+#. The configurations drop-down at the top should show all of the build targets. If not, the CMake project is probably not loaded. Go to ``File > Reload CMake Project``. The configurations should be populated
+
+Building with CLion
+###################
+
+- To build all targets, navigate to ``Build > Build All in 'Debug'``. Check that the build command displayed in the Messages window is running the correct cmake executable from your conda installation.
+- To build a specific target, select it in the configurations drop-down menu and click the hammar icon next to it.
```
Comment (sf1919) [inline dev-docs/source/CLion.rst:100]:
```suggestion
- To build a specific target, select it in the configurations drop-down menu and click the hammer icon next to it.
```

**Review by sf1919 — CHANGES_REQUESTED** [review]
Just a couple of spelling errors. Otherwise the formatting looks really good in terms of indicating the different methods for Linux and Windows.

**Review by sf1919 — APPROVED** [review]
Thanks for making the final minor changes. It looks good and the instructions work.

**Pull request merged**